nano/node/gap_cache.cpp

```diff
@@ -54,48 +54,53 @@ void nano::gap_cache::vote (std::shared_ptr<nano::vote> vote_a)
 
 			if (is_new)
 			{
-				uint128_t tally;
-				for (auto & voter : existing->voters)
-				{
-					tally += node.ledger.weight (transaction, voter);
-				}
-				bool start_bootstrap (false);
-				if (!node.flags.disable_lazy_bootstrap)
+				bootstrap_check (transaction, existing->voters, hash);
+			}
+		}
+	}
+}
+
+void nano::gap_cache::bootstrap_check (nano::transaction const & transaction_a, std::vector<nano::account> const & voters_a, nano::block_hash const & hash_a)
+{
+	uint128_t tally;
+	for (auto & voter : voters_a)
+	{
+		tally += node.ledger.weight (transaction_a, voter);
+	}
+	bool start_bootstrap (false);
+	if (!node.flags.disable_lazy_bootstrap)
+	{
+		if (tally >= node.config.online_weight_minimum.number ())
+		{
+			start_bootstrap = true;
+		}
+	}
+	else if (!node.flags.disable_legacy_bootstrap && tally > bootstrap_threshold (transaction_a))
+	{
+		start_bootstrap = true;
+	}
```

Comment:
Identical if/else in `gap_cache::vote`, so maybe extract this into `should_bootstrap` and call that from both places.